### PR TITLE
End-to-end test for Web

### PR DESCRIPTION
**Pull request by supl**
This PR adds end-to-end test for Web with audito-enabled ScalarDL.

Simply speaking, it's the web version of https://github.com/scalar-labs/scalardl-javascript-client-sdk/blob/main/node/e2e/client_service.with_auditor.test.ts

According to Jun's suggestion
https://github.com/scalar-labs/scalardl-javascript-client-sdk/pull/33#discussion_r1171997078

, this PR also moves common class files used for testing from the node folder to the common folder because they can be used for both node and web.

### CODE REVIEW

File: web/karma.conf.js [hunk 1]
```diff
@@ -7,6 +7,12 @@ module.exports = function (config) {
     // available frameworks: https://www.npmjs.com/search?q=keywords:karma-adapter
     frameworks: ['jasmine'],
 
+    client: {
+      jasmine: {
+        random: false,
```
Comment (supl) [inline web/karma.conf.js:12]: this makes karma runs test cases in order.

File: .github/workflows/web-test.yml [hunk 2]
```diff
@@ -28,6 +28,39 @@ jobs:
...
           registry-url: 'https://registry.npmjs.org'
 
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+
+      - name: Start up a Local ScalarDL Network
```
Comment (supl) [inline .github/workflows/web-test.yml:38]: Prepare a local ScalarDL network for the end-to-end test.

File: web/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "@types/web": "^0.0.96",
     "@typescript-eslint/eslint-plugin": "^5.49.0",
     "@typescript-eslint/parser": "^5.49.0",
+    "arraybuffer-loader": "^1.0.8",
```
Comment (supl) [inline web/package.json:28]: this dependency is `save-dev` only. It's for loading the class files on the browser.

File: web/client_service.ts [hunk 4]
```diff
@@ -52,7 +52,8 @@ export class ClientService extends ClientServiceBase {
       },
       protobuf as never,
       properties,
-      metadata as never
+      metadata as never,
+      true
```
Comment (supl) [inline web/client_service.ts:56]: This flag tells the base class that we are using the browser runtime.

File: web/__tests__/e2e/fixture/DeprecatedStateReader.class
Comment (jnmt) [inline web/__tests__/e2e/fixture/DeprecatedStateReader.class:1]: These contract classes are completely the same as the ones (node/e2e/fixture/*.classs) for node? If so, how about having them in the common? Is referring them from node/web something odd?

Comment (supl) [reply web/__tests__/e2e/fixture/DeprecatedStateReader.class:1]: Yes. They are identical.
I think it's a good suggestion.
Let me move them to the common folder and create soft links in the node and web folders.

Comment (supl) [reply web/__tests__/e2e/fixture/DeprecatedStateReader.class:1]: I did it in https://github.com/scalar-labs/scalardl-javascript-client-sdk/pull/33/commits/025ba30b3e92738bf8899b4bb18e610d4d314503

Please take a look 🙇

**Review by jnmt — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**